src/plugins/allHooks.js

```diff
@@ -16,8 +16,20 @@ class AllHooksPlugin {
       })
       .fromPairs()
       .value();
+    this.meta = {
```

Comment:
@kolanos this PR is getting really close, I was just wondering why you went with a function instead of a plain object like this. Since the function just exports a static object, seems more performant/straightforward to use the object directly?

Reply:
@coreylight There are a few reasons I like the meta method instead of just making it a property...

1. It isn't necessary to construct the object.
2. We could/should have a base plugin class that every class must be a prototype, in which a meta method could be required. This internalizes this check, whereas a property would need to be checked externally. I'm currently checking for a `.meta` function in the report class, but this could be done in a base class.
3. No potential issues with objects changing post-instantiation, the meta method returns a new object each time, no copies need to be made to preserve state, etc.
4. It's functional.

Thoughts?